inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp

```diff
@@ -140,6 +141,47 @@ ModelPtr FrontEnd::buildInitialModel(ie::ICNNNetwork& network) {
     return runCommonPasses(network);
 }
 
+bool FrontEnd::isLayerSupported(const std::string& type) {
+    return parsers.count(type) != 0;
+}
+
+ie::ICNNNetwork::Ptr FrontEnd::convertNetwork(ie::ICNNNetwork& network) {
+        std::shared_ptr<ie::ICNNNetwork> convertedNetwork;
+        // disable transformations for some cases
+        const auto transformationsPredicate = [](const std::shared_ptr<const ngraph::Node> &node) -> bool {
+            const bool casesWithDynamicOrStaticUsage = std::dynamic_pointer_cast<const ngraph::opset3::Gelu>(node) ||
+                                                        std::dynamic_pointer_cast<const ngraph::opset4::SoftPlus>(node);
+
+            const bool casesWithOnlyDynamicUsage = (std::dynamic_pointer_cast<const ngraph::opset3::MatMul>(node) ||
+                                                    std::dynamic_pointer_cast<const ngraph::opset3::StridedSlice>(node)) &&
+                    std::dynamic_pointer_cast<const ngraph::vpu::op::DynamicShapeResolver>(node->input_value(0).get_node_shared_ptr());
+
+            return casesWithDynamicOrStaticUsage || casesWithOnlyDynamicUsage;
+        };
+
+        auto nGraphFunc = network.getFunction();
+        // Disable shape inference (WA for generic operations)
+        ngraph::op::GenericIE::DisableReshape noReshape(nGraphFunc);
+
+        ngraph::pass::Manager manager;
+
+        manager.register_pass<ngraph::pass::CommonOptimizations>();
```

Comment:
@adudchen @Maxim-Doronin @andrejsokolov 

Why it is there? Seems like calling `CommonOptimizations` breaks Faster R-CNN support (possibly including release branch)